data_source/gateway.go

```diff
@@ -27,14 +27,152 @@ func (g *Gateway) GetBlockByNumber(blockNumber uint64) (*core.Block, error) {
 
 // GetTransaction gets the transaction for a given transaction hash from the feeder gateway,
 // then adapts it to the appropriate core.Transaction types.
-func (g *Gateway) GetTransaction(transactionHash *felt.Felt) (*core.Transaction, error) {
-	return nil, errors.New("not implemented")
+func (g *Gateway) GetTransaction(transactionHash *felt.Felt) (core.Transaction, error) {
+	response, err := g.client.GetTransaction(transactionHash)
+	if err != nil {
+		return nil, err
+	}
+
+	declareTx, deployTx, invokeTx, err := adaptTransaction(response, g)
+	if err != nil {
+		return nil, err
+	}
+
+	if declareTx != nil {
+		return declareTx, nil
+	}
+
+	if deployTx != nil {
+		return deployTx, nil
+	}
+
+	return invokeTx, nil
+}
+
+func adaptTransaction(response *clients.TransactionStatus, g *Gateway) (*core.DeclareTransaction, *core.DeployTransaction, *core.InvokeTransaction, error) {
```

Comment:
Do you really need to return one for all TX types here? Returning a single `core.Transaction` should be enough, AFAICS.